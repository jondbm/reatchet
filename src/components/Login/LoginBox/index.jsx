import React from 'react'
import styled from 'styled-components'

import Logo from '../../../assets/Logo'
import theme from '../../../theme'

/*
*****************
STYLED COMPONENTS
*****************
*/

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: ${theme.spacing.md};
  background-color: ${theme.colours.white};
  width: 420px;
  line-height: 1.42;
`

const StyledLogo = styled(Logo)`
  height: 32px;
  margin-bottom: ${theme.spacing.sm};
`

/*
*********
COMPONENT
*********
*/

const LoginBox = props => {
  const { children, className } = props
  return (
    <Wrapper className={className}>
      <StyledLogo fill={theme.colours.steel} />
      {children}
    </Wrapper>
  )
}

export default LoginBox
