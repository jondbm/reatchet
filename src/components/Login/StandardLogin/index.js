import { connect } from 'react-redux'

import StandardLogin from './StandardLogin'
import * as actions from '../../../reducers/user/actions'

export const mapStateToProps = state => ({
  hasInvalidCredentials: state.user.hasInvalidCredentials
})

export default connect(mapStateToProps, { ...actions })(StandardLogin)
