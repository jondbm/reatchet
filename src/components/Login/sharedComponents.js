import styled from 'styled-components'

import Recaptcha from '../common/Recaptcha'
import theme from '../../theme'

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  font-family: Lato, sans-serif;
  max-width: 100%;
`

export const Title = styled.span`
  text-align: center;
  font-size: 20px;
  font-weight: bold;
  color: ${theme.colours.steel};
  margin-bottom: ${theme.spacing.sm};
  max-width: 100%;
`
export const Subtitle = styled.span`
  font-size: 16px;
  text-align: center;
  color: ${theme.colours.greyDark};
  margin-bottom: ${theme.spacing.sm};
  max-width: 100%;
`

export const StyledRecaptcha = styled(Recaptcha)`
  margin-top: ${theme.spacing.xs};
`
