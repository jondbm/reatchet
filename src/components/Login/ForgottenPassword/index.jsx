import React, { Component } from 'react'
import styled from 'styled-components'

import TextInput from '../../common/TextInput'
import Button from '../../common/Buttons/Button'
import { Wrapper, Title, Subtitle, StyledRecaptcha } from '../sharedComponents'

import { validateEmail } from '../../../helpers/validation'
import theme from '../../../theme'

/*
*****************
STYLED COMPONENTS
*****************
*/

const Footer = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-end;
  align-self: stretch;
  margin-top: ${theme.spacing.sm};
`

const SubmitButton = styled(Button)`
  margin-left: ${theme.spacing.xs};
`

/*
*********
COMPONENT
*********
*/

class ForgottenPassword extends Component {
  state = {
    email: '',
    emailIsValid: true,
    recaptchaToken: ''
  }

  updateEmail = value => {
    const isValid = validateEmail(value)

    this.setState({
      email: value,
      emailIsValid: isValid
    })
  }

  handleRecaptcha = token => {
    this.setState({ recaptchaToken: token })
  }

  handleExpiredRecaptcha = () => {
    this.setState({ recaptchaToken: '' })
  }

  // can't seem to spy on history.goBack()
  /* istanbul ignore next */
  handleCancel = () => {
    this.props.history.goBack()
  }

  render() {
    const { email, emailIsValid } = this.state
    return (
      <Wrapper>
        <Title>Forgotten your password?</Title>
        <Subtitle>
          Pop your email address in below and in return weâ€™ll send you a
          password reset link!
        </Subtitle>
        <TextInput
          label="Email address"
          handleChange={this.updateEmail}
          value={email}
          isValid={emailIsValid}
          validationMessage="Enter a valid email address"
        />
        <StyledRecaptcha
          handleRecaptcha={this.handleRecaptcha}
          handleExpiredRecaptcha={this.handleExpiredRecaptcha}
          id="forgotten-password"
        />
        <Footer>
          <Button
            data-el="cancel-button"
            variant="secondary"
            onClick={this.handleCancel}
          >
            Cancel
          </Button>
          <SubmitButton data-el="submit-button">Submit</SubmitButton>
        </Footer>
      </Wrapper>
    )
  }
}

export default ForgottenPassword
