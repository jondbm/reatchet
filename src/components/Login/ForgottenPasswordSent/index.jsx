import React from 'react'
import styled from 'styled-components'

import Button from '../../common/Buttons/Button'
import LinkButton from '../../common/Buttons/LinkButton'
import { Wrapper, Title, Subtitle } from '../sharedComponents'
import EnvelopeImage from '../../../assets/EnvelopeImage'

import theme from '../../../theme'

/*
*****************
STYLED COMPONENTS
*****************
*/

const Image = styled(EnvelopeImage)`
  width: 70%;
  margin-bottom: ${theme.spacing.sm};
`

const EmailLink = styled.a`
  text-decoration: none;
  color: ${theme.colours.teal};
`

const Footer = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-end;
  align-self: stretch;
  margin-top: ${theme.spacing.sm};
`

const SubmitButton = styled(Button)`
  margin-left: ${theme.spacing.xs};
`

/*
*********
COMPONENT
*********
*/

const ForgottenPasswordSent = () => {
  return (
    <Wrapper>
      <Title>Check your mail!</Title>
      <Image />
      <Subtitle>
        We have sent you an activation email with instructions to set up your account. Don't forget
        to check your spam folder and contact{' '}
        <EmailLink href="mailto:support@seenit.io">support@seenit.io</EmailLink> if you have any
        trouble!
      </Subtitle>
      <Footer>
        <LinkButton to="/login" data-el="cancel-button" variant="secondary" onClick={() => {}}>
          Back
        </LinkButton>
        <SubmitButton data-el="submit-button">Resend</SubmitButton>
      </Footer>
    </Wrapper>
  )
}

export default ForgottenPasswordSent
