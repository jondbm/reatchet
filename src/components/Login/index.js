import { connect } from 'react-redux'

import Login from './Login'

export const mapStateToProps = state => ({
  isLoggedIn: state.user.isLoggedIn
})

export default connect(mapStateToProps, {})(Login)
