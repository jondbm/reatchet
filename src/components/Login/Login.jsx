import React, { Component } from 'react'
import { Switch, Route, Redirect } from 'react-router-dom'
import styled from 'styled-components'

import backgroundImage from '../../assets/login-background.jpg'
import LoginBox from './LoginBox'
import StandardLogin from './StandardLogin'
import ForgottenPassword from './ForgottenPassword'
import ForgottenPasswordSent from './ForgottenPasswordSent'

/*
*****************
STYLED COMPONENTS
*****************
*/

const Wrapper = styled.div`
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
`

const Background = styled.div`
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 1;
  background-image: url(${backgroundImage});
  background-size: cover;
`

const Content = styled.div`
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 2;
  display: flex;
  align-items: center;
  justify-content: center;
`

/*
*********
COMPONENT
*********
*/

class Login extends Component {
  render() {
    if (this.props.isLoggedIn) {
      let destination = '/'
      if (this.props.location.state && this.props.location.state.referrer) {
        destination = this.props.location.state.referrer
      }
      return <Redirect data-el="redirect-home" to={destination} />
    }

    return (
      <Wrapper data-el="login-page">
        <Background />
        <Content>
          <LoginBox>
            <Switch>
              <Route
                path="/login/forgotten-password/sent"
                component={ForgottenPasswordSent}
              />
              <Route
                path="/login/forgotten-password"
                component={ForgottenPassword}
              />
              <Route path="/login" component={StandardLogin} />
            </Switch>
          </LoginBox>
        </Content>
      </Wrapper>
    )
  }
}

export default Login
