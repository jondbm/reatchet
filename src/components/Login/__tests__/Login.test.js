import React from 'react'
import 'jest-styled-components'
import { MemoryRouter } from 'react-router-dom'
import { mount } from 'enzyme'

import Login from '../Login'

describe('<Login /> component', () => {
  describe('!isLoggedIn', () => {
    let component

    beforeEach(() => {
      component = mount(
        <MemoryRouter keyLength={0}>
          <Login isLoggedIn={false} />
        </MemoryRouter>
      )
    })

    it('should render the login page when not logged in', () => {
      const page = component.find('[data-el="login-page"]')
      expect(page.exists()).toBe(true)
    })

    it('should match the snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('isLoggedIn redirect', () => {
    it('should redirect once logged in', () => {
      const component = mount(
        <MemoryRouter keyLength={0}>
          <Login isLoggedIn location={{ state: { referrer: '/test' } }} />
        </MemoryRouter>
      )
      const redirect = component.find('[data-el="redirect-home"]')
      expect(redirect.exists()).toBe(true)
    })

    it('should redirect to referrer destination when one is set', () => {
      const component = mount(
        <MemoryRouter keyLength={0}>
          <Login isLoggedIn location={{ state: { referrer: '/test' } }} />
        </MemoryRouter>
      )
      const redirect = component.find('[data-el="redirect-home"]')
      expect(redirect.prop('to')).toEqual('/test')
    })

    it('should redirect to "/" when no referrer is set', () => {
      const component = mount(
        <MemoryRouter keyLength={0}>
          <Login isLoggedIn location={{}} />
        </MemoryRouter>
      )

      const redirect = component.find('[data-el="redirect-home"]')
      expect(redirect.prop('to')).toEqual('/')
    })
  })
})
