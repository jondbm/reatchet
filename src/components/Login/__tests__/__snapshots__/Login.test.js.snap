// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> component !isLoggedIn should match the snapshot 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 48px;
  background-color: #ffffff;
  width: 420px;
  line-height: 1.42;
}

.c4 {
  height: 32px;
  margin-bottom: 24px;
}

.c0 {
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
}

.c1 {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 1;
  background-image: url(login-background.jpg);
  background-size: cover;
}

.c2 {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Login
      isLoggedIn={false}
    >
      <Login__Wrapper
        data-el="login-page"
      >
        <div
          className="c0"
          data-el="login-page"
        >
          <Login__Background>
            <div
              className="c1"
            />
          </Login__Background>
          <Login__Content>
            <div
              className="c2"
            >
              <LoginBox>
                <LoginBox__Wrapper>
                  <div
                    className="c3"
                  >
                    <LoginBox__StyledLogo
                      fill="#16222a"
                    >
                      <Logo
                        className="c4"
                        fill="#16222a"
                      >
                        <svg
                          className="c4"
                          fill="#16222a"
                          viewBox="0 0 194 40"
                        >
                          <polygon
                            points="42.15 16.64 30.53 2.31 11.31 2.31 31.92 16.64 42.15 16.64"
                          />
                          <polygon
                            points="0 25.68 11.62 40 30.97 40 10.36 25.68 0 25.68"
                          />
                          <polygon
                            points="0.83 16.08 33.1 38.51 41.38 26.24 9.11 3.8 0.83 16.08"
                          />
                          <path
                            d="M66.58,33.76a14.9,14.9,0,0,1-5.25-1,16.22,16.22,0,0,1-4.55-2.53l1.36-1.83a16.43,16.43,0,0,0,4.13,2.32,12.42,12.42,0,0,0,4.45.82,6.76,6.76,0,0,0,4.1-1.22A3.9,3.9,0,0,0,72.49,27v-.09A3.06,3.06,0,0,0,72,25.14a4.75,4.75,0,0,0-1.41-1.31,9.21,9.21,0,0,0-2-.94q-1.13-.37-2.39-.75-1.45-.42-2.93-.89a11.75,11.75,0,0,1-2.65-1.19,6.16,6.16,0,0,1-1.9-1.8A4.66,4.66,0,0,1,58,15.6V15.5a5.9,5.9,0,0,1,.59-2.64,6.5,6.5,0,0,1,1.62-2.08A7.36,7.36,0,0,1,62.65,9.4a9.91,9.91,0,0,1,3.19-.49,14.67,14.67,0,0,1,4.43.72,15,15,0,0,1,4.06,2L73.1,13.5a15.89,15.89,0,0,0-3.54-1.73,11.64,11.64,0,0,0-3.82-.66,6.12,6.12,0,0,0-4,1.19,3.67,3.67,0,0,0-1.48,3v.09a2.7,2.7,0,0,0,.56,1.71,4.83,4.83,0,0,0,1.48,1.24,11.84,11.84,0,0,0,2.13.91q1.22.4,2.48.77,1.45.42,2.86.91a10.68,10.68,0,0,1,2.53,1.26A6.36,6.36,0,0,1,74.16,24a4.9,4.9,0,0,1,.68,2.67v.09a6.35,6.35,0,0,1-2.39,5.08,8.14,8.14,0,0,1-2.63,1.4,10.35,10.35,0,0,1-3.24.49"
                          />
                          <path
                            d="M101,20.21a13.62,13.62,0,0,0-.73-3.49,9.72,9.72,0,0,0-1.64-3A7.87,7.87,0,0,0,96,11.7a8.56,8.56,0,0,0-7-.07,8.35,8.35,0,0,0-2.7,1.95,10.54,10.54,0,0,0-1.9,3,11,11,0,0,0-.89,3.68ZM92.87,33.85A11.57,11.57,0,0,1,88.38,33a11.37,11.37,0,0,1-3.76-2.53,12,12,0,0,1-2.56-3.94,13.37,13.37,0,0,1-.94-5.11v-.09A13.49,13.49,0,0,1,82,16.43a12.46,12.46,0,0,1,2.39-4A11.39,11.39,0,0,1,88,9.8a10.31,10.31,0,0,1,4.5-1,10.44,10.44,0,0,1,4.55,1,10.24,10.24,0,0,1,3.45,2.65,12.08,12.08,0,0,1,2.2,4,15.26,15.26,0,0,1,.77,4.95v.49a2.94,2.94,0,0,1,0,.45H83.56a10.42,10.42,0,0,0,1,4,10.29,10.29,0,0,0,2.11,2.94A8.71,8.71,0,0,0,89.55,31a9.33,9.33,0,0,0,3.41.63,9.74,9.74,0,0,0,4.77-1.08,14.08,14.08,0,0,0,3.46-2.67l1.59,1.41a16.68,16.68,0,0,1-4.16,3.26,11.74,11.74,0,0,1-5.76,1.29"
                          />
                          <path
                            d="M130.49,20.21a13.62,13.62,0,0,0-.73-3.49,9.72,9.72,0,0,0-1.64-3,7.88,7.88,0,0,0-2.61-2.06,8.56,8.56,0,0,0-7-.07,8.35,8.35,0,0,0-2.7,1.95,10.54,10.54,0,0,0-1.9,3,11,11,0,0,0-.89,3.68Zm-8.13,13.64a11.57,11.57,0,0,1-4.49-.89,11.37,11.37,0,0,1-3.76-2.53,12,12,0,0,1-2.56-3.94,13.36,13.36,0,0,1-.94-5.11v-.09a13.49,13.49,0,0,1,.87-4.85,12.46,12.46,0,0,1,2.39-4,11.38,11.38,0,0,1,3.59-2.67,10.31,10.31,0,0,1,4.5-1,10.44,10.44,0,0,1,4.55,1A10.24,10.24,0,0,1,130,12.42a12.08,12.08,0,0,1,2.2,4,15.26,15.26,0,0,1,.77,4.95v.49a2.93,2.93,0,0,1,0,.45H113.05a10.4,10.4,0,0,0,1,4,10.29,10.29,0,0,0,2.11,2.94A8.71,8.71,0,0,0,119,31a9.33,9.33,0,0,0,3.41.63,9.74,9.74,0,0,0,4.77-1.08,14.08,14.08,0,0,0,3.46-2.67l1.59,1.41a16.68,16.68,0,0,1-4.16,3.26,11.74,11.74,0,0,1-5.76,1.29"
                          />
                          <path
                            d="M140,9.38h2.3v4.45a11.14,11.14,0,0,1,3.28-3.56,9.08,9.08,0,0,1,5.34-1.45,10.61,10.61,0,0,1,4,.73,8.43,8.43,0,0,1,4.9,5.16,11.9,11.9,0,0,1,.63,4V33.29h-2.3V19.08a8.63,8.63,0,0,0-2-5.93,7.11,7.11,0,0,0-5.63-2.23,8.35,8.35,0,0,0-3.28.63,8.15,8.15,0,0,0-2.63,1.76,8,8,0,0,0-1.76,2.7,9.14,9.14,0,0,0-.63,3.45V33.29H140Z"
                          />
                          <path
                            d="M170.7,33.29H173V9.38h-2.3Zm-.28-30.15h2.86V0h-2.86Z"
                          />
                          <path
                            d="M190,33.71a8.91,8.91,0,0,1-2.63-.38,5.51,5.51,0,0,1-3.56-3.26,7.6,7.6,0,0,1-.52-2.93V11.49h-3.47V9.38h3.47V1.73h2.3V9.38H194v2.11h-8.39V26.87a4.29,4.29,0,0,0,1.34,3.6,5.48,5.48,0,0,0,3.45,1,7.72,7.72,0,0,0,1.69-.18,10.53,10.53,0,0,0,1.83-.62v2.16a10,10,0,0,1-1.85.63,9.1,9.1,0,0,1-2,.21"
                          />
                        </svg>
                      </Logo>
                    </LoginBox__StyledLogo>
                    <Switch />
                  </div>
                </LoginBox__Wrapper>
              </LoginBox>
            </div>
          </Login__Content>
        </div>
      </Login__Wrapper>
    </Login>
  </Router>
</MemoryRouter>
`;
